libexec/bats-core/bats

```diff
@@ -112,9 +117,18 @@ for filename in "${arguments[@]}"; do
 
   if [[ -d "$filename" ]]; then
     shopt -s nullglob
-    for suite_filename in "$filename"/*.bats; do
-      filenames["${#filenames[@]}"]="$suite_filename"
-    done
+    if [[ "$recursive" -eq 1 ]]; then
+      OIFS="${IFS}"
```

Comment:
Agreed, but there are other locations in the code where this is done, and I wanted to make sure I wasn't intentionally missing a bug in an older Bash version somewhere by not including that.  It can be removed if desired.

Reply:
More specifically, libexec/bats-exec-test, in the run() function:

```bash
run() {
  local origFlags="$-"
  set +eET
  local origIFS="$IFS"
  output="$("$@" 2>&1)"
  status="$?"
  IFS=$'\n' lines=($output)
  IFS="$origIFS"
  set "-$origFlags"
}
```

Reply:
Yeah, it's needed in `run()`, because the `IFS=$'\n' line=($output)` line acts as two assignments, not as a command-specific override. But `while IFS= read` is safe, so please remove the `OIFS` lines. (Of course, run the tests to confirm!)

Reply:
It doesn't appear to break the tests, so I've removed them.  I had some concern since I was using process substitution to keep variable changes within the current shell's scope that IFS might get clobbered.  Glad to see that is not the case!